client.go

```diff
@@ -0,0 +1,75 @@
+package nomadclient
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	dphttp "github.com/ONSdigital/dp-net/http"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+type Nomad struct {
+	Client dphttp.Clienter
+	URL string
+	// Name needed for checker later.
+	// Name string
+}
+
+const prefix = "https://"
+var caCertPool *x509.CertPool
+
+func NewClient(nomadEndpoint, nomadCACert string, nomadTLSSkipVerify bool) (*Nomad, error){
+
+	dpHTTPClient := *dphttp.DefaultClient
+
+	if strings.HasPrefix(nomadEndpoint, prefix) {
+		tlsConfig, err := createTlsConfig(nomadCACert, nomadTLSSkipVerify)
+		if err != nil {
+			return nil, err
+		}
+
+		dpHTTPClient.HTTPClient.Transport = &http.Transport{TLSClientConfig: tlsConfig}
+	}
+	return  &Nomad{
+		Client: &dpHTTPClient,
+		URL:    nomadEndpoint,
+		// Name:   name,
+	}, nil
+}
+
+func createTlsConfig (nomadCACert string, nomadLSSkipVerify bool) (*tls.Config, error) {
```

Comment:
function name should be `createTLSConfig` acronyms in the middle of function names are capitalised